chain/chain/src/chain.rs

```diff
@@ -3018,6 +3018,30 @@ impl Chain {
             .ok_or_else(|| ErrorKind::DBNotFoundErr(format!("EXECUTION OUTCOME: {}", id)).into())
     }
 
+    pub fn retrieve_headers(
```

Comment:
please add comments on what this function does (as  I thought that it would return the block headers for these hashes, but from what I see it returns headers for the chain that starts from the lowest hash -- quite confusing)